src/integrators/integrator_utils.jl

```diff
@@ -150,7 +150,15 @@ end
       integrator.tprev = integrator.t
       if typeof(integrator.t)<:AbstractFloat && !isempty(integrator.opts.tstops)
         tstop = integrator.tdir * top(integrator.opts.tstops)
-        @fastmath abs(ttmp - tstop) < 10eps(integrator.t) ? (integrator.t = tstop) : (integrator.t = ttmp)
+        if @fastmath abs(ttmp - tstop) < 10eps(integrator.t)
+          integrator.t = tstop
+          integrator.W.curt = tstop
```

Comment:
Setting `integrator.W.curt` in here won't work  because `W.curt` is increased in https://github.com/SciML/DiffEqNoiseProcess.jl/blob/b8c8587e8a1b7ab9b59238e35f3e7b1cc0af7228/src/noise_interfaces/noise_process_interface.jl#L13